### PR TITLE
Hookup autocomplete

### LINKED ISSUE #91: Hook-up exhibit item autocomplete in Mirador widget

### PR DESCRIPTION
**Pull request by mejackreed**
Closes #91 

Sometimes it seems like that things aren't loading, its just waiting on the BAV manifests/thumbs. #105 could help ??

autocomplete-from-exhibit

There are some overridden methods here, all documented though I'm not sure this is the most "sir trevor" way of doing things.

### CODE REVIEW

File: app/assets/javascripts/mirador_widget_admin.js [hunk 1]
```diff
@@ -128,6 +128,10 @@
         showSourceLocationInput(block, sourceLocationValue(block));
       },
 
+      updateHiddenMiradorConfig(block) {
+        updateHiddenMiradorConfig(block);
+      },
```
Comment (jkeck) [inline app/assets/javascripts/mirador_widget_admin.js]: ❓

File: app/assets/javascripts/zpotlight/blocks/mirador_block.js [hunk 2]
```diff
@@ -22,13 +34,82 @@ SirTrevor.Blocks.Mirador = (function() {
...
+    afterPanelDelete: function() {
+      var block = $(this).closest('[data-behavior="mirador-widget"]');
+      var panel = $(this).closest('.field');
+      block.trigger('item-removed', { block: block, panel: panel });
+    },
+
+    /**
+     * A simplification from https://github.com/projectblacklight/spotlight/blob/master/app/assets/javascripts/spotlight/blocks/resources_block.js
+     */
+    _itemPanel: function(data) {
```
Comment (jkeck) [inline app/assets/javascripts/zpotlight/blocks/mirador_block.js]: This seems nearly identical to the code over in `MiradorWidgetAdmin` https://github.com/sul-dlss/vatican_exhibits/blob/cb212b58e7b2939fc43c13888dd4957d4ded8cff/app/assets/javascripts/mirador_widget_admin.js#L148-L167

Should these be re-using each-other in someway?

File: app/assets/javascripts/mirador_widget_admin.js (hunk 1 above)
Comment (mejackreed) [reply app/assets/javascripts/mirador_widget_admin.js]: Yeah, just making the available globally.. could easily be refactored I guess.

Comment (jkeck) [reply app/assets/javascripts/mirador_widget_admin.js]: Oh, I see, I needed to see the diff above what GH was showing to see that this was in the returned object.  I would be 👌w/ just making the function public, but am totally fine with this too (revealing module pattern, I 👀 you).

Comment (mejackreed) [reply app/assets/javascripts/mirador_widget_admin.js]: I've made the method public in b1c9cf4

File: spec/support/selenium_webdriver.rb [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
 Capybara.register_driver :headless_chrome do |app|
   capabilities = Selenium::WebDriver::Remote::Capabilities.chrome(
-    chromeOptions: { args: %w[headless disable-gpu no-sandbox] },
+    chromeOptions: { args: %w[headless disable-gpu no-sandbox window-size=1280,1696] },
```
Comment (camillevilla) [inline spec/support/selenium_webdriver.rb]: Neat! 🤓

File: app/assets/javascripts/zpotlight/blocks/mirador_block.js [hunk 4]
```diff
@@ -46,15 +76,15 @@ SirTrevor.Blocks.Mirador = (function() {
...
+              '<input type="radio" name="<%= blockID %>_source_location" id="<%= blockID + "_source_location_exhibit_label" %>" data-behavior="source-location-select" value="exhibit" checked> <%= i18n.t("blocks:mirador:source_location:exhibit:label") %>',
             '</label>',
             '<label for="<%= blockID + "_source_location_iiif_label"  %>" class="radio-inline">',
               '<input type="radio" name="<%= blockID %>_source_location" id="<%= blockID + "_source_location_iiif_label" %>" data-behavior="source-location-select" value="iiif"> <%= i18n.t("blocks:mirador:source_location:iiif:label") %>',
             '</label>',
           '</div>',
           '<div class="form-inline" data-source-location="exhibit">',
             '<div class="form-group">',
-              '<input type="text" placeholder="<%= i18n.t("blocks:mirador:source_location:exhibit:placeholder") %>" data-behavior="source-location-input" />',
+              '<input type="text" class="st-input-string form-control item-input-field" data-twitter-typeahead="true" placeholder="<%= i18n.t("blocks:mirador:source_location:exhibit:placeholder")%>"/>',
```
Comment (camillevilla) [inline app/assets/javascripts/zpotlight/blocks/mirador_block.js]: Can we add a `label` to this for screenreader folks? I'll add one to the `IIIF manifest` version of the form in #99.

Comment (mejackreed) [reply app/assets/javascripts/zpotlight/blocks/mirador_block.js]: Sure thing. Added in latest commit. FWIW, I'm not sure that we have a label upstream in Spotlight? https://github.com/projectblacklight/spotlight/blob/master/app/assets/javascripts/spotlight/block_mixins/autocompleteable.js#L34

Or even a larger question, if the administrative and curatorial screens (mainly SirTrevor) has been evaluated for accessibility. On a quick try, I couldn't even change the SirTrevor widget type using keyboard only commands. Thanks for raising the issue!

**Pull request merged**